src/app/client/src/app/modules/public/module/player/components/public-collection-player/public-collection-player.component.ts

```diff
@@ -330,7 +330,21 @@ export class PublicCollectionPlayerComponent implements OnInit, OnDestroy, After
       sessionStorage.setItem('singleContentRedirect', 'singleContentRedirect');
       this.router.navigate(['/get/dial/', this.dialCode]);
     } else {
-      this.navigationHelperService.navigateToPreviousUrl('/explore');
+      if (this.isOffline) {
```

Comment:
const  previousUrl =  this.navigationHelperService.getPreviousUrl();
       if (Boolean(_.includes(previousUrl.url, '/play/collection/'))) 
       return this.router.navigate(['/']);

previousUrl.queryParams ? this.router.navigate([previousUrl.url], {queryParams: previousUrl.queryParams}): this.router.navigate([previousUrl.url]);
    
      } else {
        this.navigationHelperService.navigateToPreviousUrl('/explore');
      }